Intermittent test failure - org.mozilla.fenix.tabstray.NavigationInteractorTest - onOpenRecentlyClosedClicked calls navigation on DefaultNavigationInteractor

### Firebase Test Run:
https://treeherder.mozilla.org/logviewer?job_id=336430022&repo=fenix
### Stacktrace:
```
task 2021-04-14T01:04:54.184Z] SUITE: org.mozilla.fenix.tabstray.NavigationInteractorTest
[task 2021-04-14T01:04:54.381Z]   TEST: onOpenRecentlyClosedClicked calls navigation on DefaultNavigationInteractor
[task 2021-04-14T01:04:54.981Z]   SUCCESS
[task 2021-04-14T01:04:54.982Z] 
[task 2021-04-14T01:04:54.982Z]   TEST: onBookmarkTabs calls navigation on DefaultNavigationInteractor
[task 2021-04-14T01:04:56.082Z]   FAILURE
[task 2021-04-14T01:04:56.082Z] 
[task 2021-04-14T01:04:56.082Z] java.lang.AssertionError: Verification failed: call 2 of 2: AddBookmarksUseCase(child of #1093#1096).invoke(any(), any(), any(), any())) was not called
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:66)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.MockKDsl.internalCoVerify(API.kt:143)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.MockKKt.coVerify(MockK.kt:169)
[task 2021-04-14T01:04:56.082Z] 	at io.mockk.MockKKt.coVerify$default(MockK.kt:166)
[task 2021-04-14T01:04:56.082Z] 	at org.mozilla.fenix.tabstray.NavigationInteractorTest.onBookmarkTabs calls navigation on DefaultNavigationInteractor(NavigationInteractorTest.kt:182)
[task 2021-04-14T01:04:56.082Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[task 2021-04-14T01:04:56.082Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[task 2021-04-14T01:04:56.082Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[task 2021-04-14T01:04:56.082Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
[task 2021-04-14T01:04:56.082Z] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
[task 2021-04-14T01:04:56.083Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
[task 2021-04-14T01:04:56.083Z] 	at sun.reflect.GeneratedMethodAccessor179.invoke(Unknown Source)
[task 2021-04-14T01:04:56.083Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[task 2021-04-14T01:04:56.083Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
[task 2021-04-14T01:04:56.083Z] 	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
[task 2021-04-14T01:04:56.084Z] 	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:119)
[task 2021-04-14T01:04:56.084Z] 	at sun.reflect.GeneratedMethodAccessor178.invoke(Unknown Source)
[task 2021-04-14T01:04:56.084Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[task 2021-04-14T01:04:56.084Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
[task 2021-04-14T01:04:56.084Z] 	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
[task 2021-04-14T01:04:56.085Z] 	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
[task 2021-04-14T01:04:56.085Z] 	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
[task 2021-04-14T01:04:56.085Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[task 2021-04-14T01:04:56.085Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[task 2021-04-14T01:04:56.085Z] 	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
[task 2021-04-14T01:04:56.085Z] 	at java.lang.Thread.run(Thread.java:748)
```
### Build: 
https://github.com/mozilla-mobile/fenix/commit/990bfa7e6dd5894d61473a41da6129e6947974e2

Regression from #18889 ? @MarcLeclair

--- a/app/src/test/java/org/mozilla/fenix/tabstray/NavigationInteractorTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/tabstray/NavigationInteractorTest.kt
@@ -13,6 +13,8 @@ import io.mockk.mockk
 import io.mockk.verify
 import io.mockk.mockkStatic
 import io.mockk.unmockkStatic
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
 import mozilla.components.browser.state.state.BrowserState
 import mozilla.components.browser.state.state.TabSessionState
 import mozilla.components.browser.state.state.createTab as createStateTab
@@ -176,8 +178,9 @@ class NavigationInteractorTest {
         unmockkStatic("org.mozilla.fenix.collections.CollectionsDialogKt")
     }
 
+    @OptIn(ExperimentalCoroutinesApi::class)
     @Test
-    fun `onBookmarkTabs calls navigation on DefaultNavigationInteractor`() {
+    fun `onBookmarkTabs calls navigation on DefaultNavigationInteractor`() = runBlockingTest {
         navigationInteractor.onSaveToBookmarks(listOf(createTrayTab()))
         coVerify(exactly = 1) { bookmarksUseCase.addBookmark(any(), any(), any()) }
     }

